pkg/cmd/promhelperservice/handlers/instanceconfigs/helpers.go

```diff
@@ -0,0 +1,133 @@
+// Copyright 2024 The Cockroach Authors.
+
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package instanceconfigs
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+)
+
+type ResultResponse string
+
+const (
+	// resourceName and version for the API
+	resourceName    = "instance-configs"
+	resourceVersion = "v1"
+	// ClusterIDParam is the url parameter
+	ClusterIDParam = "cluster_id"
+	// defaultInstanceConfigDir is the default instance config directory from where the yml files are created/removed.
+	defaultInstanceConfigDir = "/opt/prom/prometheus/instance-configs"
+	instanceConfigFileExt    = "yml"
+	// instanceConfigEnvVar needs to be set to override the defaultInstanceConfigDir
+	instanceConfigEnvVar = "INSTANCE_CONFIG_DIR"
+
+	// Result responses
+	SUCCESS ResultResponse = "success"
+	INVALID ResultResponse = "invalid"
+	FAILED  ResultResponse = "failed"
+)
+
+// FormInstanceConfigUrl returns the POST url for instance config
+func FormInstanceConfigUrl() string {
+	return fmt.Sprintf("%s/%s", resourceVersion, resourceName)
+}
+
+// FormInsConfUrlWithClusterID returns the url with cluster ID for instance config
+func FormInsConfUrlWithClusterID() string {
+	return fmt.Sprintf("%s/%s/{%s}", resourceVersion, resourceName, ClusterIDParam)
+}
+
+// getInstanceConfigDir returns the instance config dir. The value can be read from
+// env instanceConfigEnvVar, else the defaultInstanceConfigDir is used.
+func getInstanceConfigDir() string {
+	instanceConfigDir := os.Getenv(instanceConfigEnvVar)
+	if instanceConfigDir == "" {
+		instanceConfigDir = defaultInstanceConfigDir
+	}
+	return instanceConfigDir
+}
+
+// Node has the configuration of a specific node of a cluster
+type Node struct {
+	NodeID string `json:"node_id"`
+	// Targets are the list of targets in the host:port format
+	Targets []string `json:"targets"`
+	// Labels are the labels to be added for the node. Note that the "node" label is automatically added from NodeID
+	Labels map[string]string `json:"labels"`
+}
+
+// instanceConfigRequest is the HTTP request received for generating instance config
+type instanceConfigRequest struct {
```

Comment:
This is with the same though of using this later for PUT or PATCH. This may also become a GET response. I will rename it as instanceConfig to make it generic. There is no body for DELETE.